applications/nrf_desktop/src/hw_interface/Kconfig.battery_meas

```diff
@@ -22,6 +22,13 @@ config DESKTOP_BATTERY_MEAS
 
 endchoice
 
+config DESKTOP_BATTERY_MEAS_DEF_PATH
+	string "Configuration file"
+	depends on DESKTOP_BATTERY_MEAS
+	default "battery_def.h"
```

Comment:
I don't think this is the right approach.

Header files like this usually don't belong in Kconfig.
Exception is where such a header is a configuration header like mbed TLS, where the header is a pure `#define FEATURE_X_ENABLE`

But `battery_def.h` is far from that.

Has no other approaches been considered ?
Just because you can do this through Kconfig doesn't mean it's the right approach.

Reply:
I guess we could do it in several ways. Part of the application framework we created during nrf_desktop development, the same we use on other apps at the moment as CAF, introduced headers bringing configuration to modules.
Modules are enabled with Kconfig (as are other SW components in the system). At the moment component tries to include header with its configuration (the default name is used). We would however want to allow clients (and ourselves) more flexibility in this area. This is especially true when single board has different roles.

The framework was done some time ago. Some things were not available then in Zephyr. Maybe there are some new things we can use for this purpose... Maybe you have something specific in mind.

Reply:
To a start, all those headers are already following the `<app>/configuration/<board>` scheme (correct me if i'm wrong).
https://github.com/nrfconnect/sdk-nrf/tree/e61a8352dd6b1faaa3092f536feb136ce1832c6a/applications/nrf_desktop/configuration/nrf52833dk_nrf52833

So for users making custom boards we really don't need this, cause they will simply place their custom `_def.h` files with their custom board configuration in `configuration/my_board`.

A user who wants to modify any of those `*_def.h`  files in an app like `nrf_desktop` for Nordic boards, for example `nrf52840dk` would also need to update the `prj.conf` file with this proposed Kconfig setting, or create an out-of-tree Kconfig fragment and remember to use that with `-DOVERLAY_CONFIG=<fragment>.conf`.

The first requires patching an upstream repo (seen from downstream user), the second is error prone when having to remember to use `-DOVERLAY_CONFIG`, as well as configs for the app + board is scattered in two places.

So to handle the this last use-case, then maybe it would be better if we allowed user to control this setting externally:
https://github.com/nrfconnect/sdk-nrf/blob/e61a8352dd6b1faaa3092f536feb136ce1832c6a/applications/nrf_desktop/CMakeLists.txt#L20

That way a user could say, use the `NCS` nrf_desktop, but please pickup all configs from `my_location` instead of the default `configuration/<board>`.

I also believe such a solution will benefit IDEs, as they will have much easier in looking up the correct header file, compared to looking up this file:
https://github.com/nrfconnect/sdk-nrf/blob/e61a8352dd6b1faaa3092f536feb136ce1832c6a/applications/nrf_desktop/src/hw_interface/battery_meas.c#L20

I currently don't have the insights on the exact use-cases you want to support, but I think it would be beneficial if we take a meeting to understand the use-cases anf possible solutions.

There are many other solutions possible besides the one I sketched in this comment.